xlators/mgmt/glusterd/src/glusterd-utils.c

```diff
@@ -669,11 +669,7 @@ glusterd_volinfo_unref(glusterd_volinfo_t *volinfo)
 
     pthread_mutex_lock(&conf->volume_lock);
     {
-        pthread_mutex_lock(&volinfo->reflock);
-        {
-            refcnt = --volinfo->refcnt;
-        }
-        pthread_mutex_unlock(&volinfo->reflock);
+        refcnt = GF_ATOMIC_DEC(volinfo->refcnt);
```

Comment:
nit: refcnt probably need to be an int64_t?

Reply:
Hm... `long` seems fits better because `sizeof(long)` is expected to be equal to `sizeof(gf_atomic_t)` both on 32-bit and 64-bit platforms (except pathological cases where atomics uses mutexes internally).